trin-core/src/types/accumulator.rs

```diff
@@ -0,0 +1,219 @@
+use ethereum_types::U256;
+use ssz::Encode;
+use ssz_derive::{Decode, Encode};
+use ssz_types::{typenum, VariableList};
+use tree_hash::{MerkleHasher, TreeHash};
+use tree_hash_derive::TreeHash;
+
+use crate::types::header::Header;
+
+const EPOCH_SIZE: usize = 8192; // number of blocks / epoch
+const MAX_HISTORICAL_EPOCHS: usize = 131072; // 2 ** 17
+
+/// SSZ List[Hash256, max_length = MAX_HISTORICAL_EPOCHS]
+/// List of historical epoch accumulator merkle roots preceding current epoch.
+pub type HistoricalEpochsList = VariableList<tree_hash::Hash256, typenum::U8192>;
+
+/// SSZ List[HeaderRecord, max_length = EPOCH_SIZE]
+/// List of (block_number, block_hash) for each header in the current epoch.
+pub type HeaderRecordList = VariableList<HeaderRecord, typenum::U131072>;
+
+/// SSZ Container
+#[derive(Clone, Debug, Decode, Encode, TreeHash)]
+pub struct MasterAccumulator {
```

Comment:
nit: "SSZ Container" does not describe what the `MasterAccumulator` is. I would prefer something more descriptive.